neptune/new/sync/sync.py

```diff
@@ -0,0 +1,233 @@
+    ) -> None:
+        for name in qualified_runs_names:
+            run = get_run(name, self._backend)
+            if run:
+                run_path = base_path / ASYNC_DIRECTORY / f"{run.type.value}__{run.id}"
+                run_path_deprecated = base_path / ASYNC_DIRECTORY / f"{run.id}"
+                if run_path.exists():
+                    self.sync_run(run_path=run_path, run=run)
+                elif run_path_deprecated.exists():
+                    self.sync_run(run_path=run_path_deprecated, run=run)
+                else:
+                    click.echo(
+                        "Warning: Run '{}' does not exist in location {}".format(
+                            name, base_path
+                        ),
+                        file=sys.stderr,
+                    )
+
+    def _register_offline_run(
+        self, project: Project, container_type: ContainerType
+    ) -> Tuple[Optional[ApiExperiment], bool]:
+        try:
+            if container_type == ContainerType.RUN:
+                return self._backend.create_run(project.id), True
+            else:
+                # No need for registering project.
+                # Project must've been registered before.
+                return self._backend.get_run(project.id), False
+        except Exception as e:
+            click.echo(
+                "Exception occurred while trying to create a run "
+                "on the Neptune server. Please try again later",
+                file=sys.stderr,
+            )
+            logging.exception(e)
+            return None, False
+
+    @staticmethod
+    def _move_offline_run(
+        base_path: Path, offline_dir: str, server_id: str, server_type: ContainerType
+    ) -> None:
+        online_dir = f"{server_type.value}__{server_id}"
+        # create async directory for run
+        (base_path / ASYNC_DIRECTORY / online_dir).mkdir(parents=True)
+        # mv offline directory inside async one
+        (base_path / OFFLINE_DIRECTORY / offline_dir).rename(
+            base_path / ASYNC_DIRECTORY / online_dir / "exec-0-offline"
+        )
+
+    def register_offline_runs(
+        self, base_path: Path, project: Project, offline_dirs: Iterable[str]
+    ) -> List[ApiExperiment]:
+        result = []
+        for dir_ in offline_dirs:
+            dir_path = base_path / OFFLINE_DIRECTORY / dir_
+            if dir_path.is_dir():
+                run, registered = self._register_offline_run(
+                    project, container_type=ContainerType.RUN
+                )
+                if run:
+                    self._move_offline_run(
+                        base_path=base_path,
+                        offline_dir=dir_,
+                        server_id=run.id,
+                        server_type=run.type,
+                    )
+                    verb = "registered as" if registered else "recognized as"
+                    click.echo(f"Offline run {dir_} {verb} {get_qualified_name(run)}")
```

Comment:
current master has `Offline {container_type.value} ...`, why change that since you have `run.type`?

Reply:
I didn't restore that, but still. We support only run for now.

I didn't want to worry about other types yet.